RFS/src/main/java/com/rfs/common/http/RestClient.java

```diff
@@ -0,0 +1,185 @@
+package com.rfs.common.http;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+
+import com.rfs.netty.ReadMeteringHandler;
+import com.rfs.netty.WriteMeteringHandler;
+import com.rfs.tracing.IRfsContexts;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import lombok.SneakyThrows;
+import reactor.core.publisher.Mono;
+import reactor.netty.Connection;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.client.HttpClientRequest;
+import reactor.netty.tcp.SslProvider;
+import reactor.util.annotation.Nullable;
+
+public class RestClient {
+    public static class Response {
+        public final int code;
+        public final String body;
+        public final String message;
+        public final Map<String, String> headers;
+
+        public Response(int responseCode, String responseBody, String responseMessage, Map<String, String> responseHeaders) {
+            this.code = responseCode;
+            this.body = responseBody;
+            this.message = responseMessage;
+            this.headers = responseHeaders;
+        }
+    }
+
+    public final ConnectionContext connectionContext;
+    private final HttpClient client;
+    private final AuthTransformer authTransformer;
+
+    @SneakyThrows
```

Comment:
Lets not use `@SneakyThrows` on any library systems since it hides a failure state.  If you want to catch and wrap with an exception derived from RuntimeException thats OK with me rather than a checked exception.  What do you think?

Reply:
While this @SneakyThrows was pre-existing in the RestClient, i've removed it.